compiler/rustc_builtin_macros/src/test_harness.rs

```diff
@@ -352,9 +354,12 @@ fn mk_tests_slice(cx: &TestCtxt<'_>, sp: Span) -> P<ast::Expr> {
     debug!("building test vector from {} tests", cx.test_cases.len());
     let ecx = &cx.ext_cx;
 
+    let mut tests = cx.test_cases.clone();
+    tests.sort_by(|a, b| a.name.cmp(&b.name).reverse());
```

Comment:
Is `name` should really be `String` and not `Symbol`?
It can be
```rust
tests.sort_unstable_by(|a, b| a.name.as_str().cmp(&b.name.as_str()).reverse());
```